src/coreclr/tools/superpmi/superpmi-shim-collector/icorjitinfo.cpp

```diff
@@ -115,6 +115,12 @@ CorInfoInline interceptor_ICJI::canInline(CORINFO_METHOD_HANDLE callerHnd,    /*
     return temp;
 }
 
+void interceptor_ICJI::beginInlining(CORINFO_METHOD_HANDLE inlinerHnd,
+                                     CORINFO_METHOD_HANDLE inlineeHnd)
+{
+    // do nothing
```

Comment:
```suggestion
    mc->cr->AddCall("beginInlining");
    original_ICorJitInfo->beginInlining(inlinerHnd, inlineeHnd);
```

Without this call, when doing a spmi collection (under crossgen), crossgen won't see this call from the JIT.

Normally, we'd also add `mc->cr->recBeginInlining()` and all the mechanism around saving/replaying/dumping this call. I guess it's ok we don't add that because nobody would use it -- on replay we don't need it.

Reply:
Yes, this caused superpmi collections with crossgen2 to fail.